docs/proposals/perf.md

```diff
@@ -0,0 +1,166 @@
+  * [Proposal](#proposal)
+    * [Test Framework](#test-framework)
+    * [Test Tools](#test-tools)
+    * [Metrics Tools](#metrics-tools)
+  * [Test Scenarios](#test-Scenarios)
+
+## Motivation
+
+Currently KubeEdge test is focused on automated test suites for unit, integration and e2e test and validation.
+KubeEdge allows the users to manage large scale of edge nodes, devices from cloud.
+A set of test specifically performance test can be used to determine the non-functional
+characteristics of KubeEdge such as latency, throughput, cpu usage, memory usage and so on.
+As a result, we can also evaluate the future improvement items for KubeEdge.
+
+This proposal lists the possible performance test scenarios and test cases for KubeEdge.
+
+### Goals
+
+* Benchmark the performance against the following Service Level Objectives:
+  * Latency: time cost from the moment when the server gets the request to last byte of response sent to the users.
+  * Throughput: measure how many requests can be served within given time.
+  * Scalability: potential scaling capacity under different load conditions.
+  * CPU Usage: measure the cpu usage of KubeEdge under different load conditions.
+  * Memory Usage: measure the memory usage of KubeEdge under different load conditions.
+* Performance test can be able to run against both dockerized and un-dockerized version of KubeEdge.
+
+### Non-goals
+
+* To design the specific implementation of single performance test.
+
+## Proposal
+
+### Test Framework
+<img src="../images/perf/perf-test-framework.png">
+
+### Test Tools
+* [Kubemark](https://github.com/kubernetes/kubernetes/tree/master/test/kubemark)
+* [Fortio](https://github.com/fortio/fortio)
+
+### Metrics Tools
+* [Prometheus](https://github.com/prometheus/prometheus)
+* [Grafana](https://github.com/grafana/grafana)
+
+## Test Scenarios
+
+### 1. Edge Nodes join in K8S Cluster
+<img src="../images/perf/perf-edgenodes-join-cluster.png">
+
+Test Cases:
+* Measure Edge Nodes join in K8S Cluster startup time.
+
+  Different numbers of Edge Nodes need be tested.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+  This test case ends with all Edge Nodes are in `Ready` status.
+
+### 2. Create Device/Device Model from Cloud
+<img src="../images/perf/perf-create-device.png">
+
+This scenario is expected to measure the northbound API of KubeEdge.
+
+Test Cases:
+* Measure the latency between K8S API Server and KubeEdge Cloud Part.
+* Measure the throughput between K8S API Server and KubeEdge Cloud Part.
+
+### 3. Report Device Status to Edge
+<img src="../images/perf/perf-report-devicestatus.png">
+
+This scenario is expected to measure the southbound API of KubeEdge.
+
+Test Cases:
+* Measure the latency between KubeEdge Edge Part and device.
+
+  Device numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+* Measure the throughput between KubeEdge Edge Part and device.
+
+  Device numbers are one of `[1, 10, 20, 50, 100, 200]`.
```

Comment:
Do you mean register devices to the same node at the same time?

Reply:
yup